### DISCUSSION TITLE
Some UX tweaks that would improve the editor

### DISCUSSION THREAD

**Discussion by PiorGit** [TheHerobrine/DarkerNodes#26] [General]
Hello Hugo, thank you for your work and contributions.

DarkerNodes and ElectronicNodes do look great, and they make me wonder if there are other ways to tweak the feel of the editor beyond just the looks. Here are a few things that I believe could be made smoother : 

- Selecting objects in the viewport happens on click-release rather than on-click. This alone makes the editor feel way more sluggish than it needs to be.
- Editable input fields/highlighted text seem to behave in a non-standard way leading to many input errors when renaming things or even merely typing in values. I feel like this was designed in the interest of speed (reducing double clicks) but it leads to a very "unstable" behavior. As often with this sort of things, Blender tends to exhibit a much more robust behavior in that regard for instance.
- Rollouts allowing to pick something from a long list (for instance, selecting a bone to parent a BP component to) don't allow to just hit enter to confirm after filtering for a name.
- The right-click search box for BP nodes could be completely overhauled/simplified to be much clearer and more flexible.

... and so  many other things :)

These obviously go beyond the scope of the DN plugin as it currently is but I think these would be great areas to investigate, at the very least to find out if such changes are even possible. I would love to hear your thoughts on this !

**Comment by hugoattal** [comment]
Hey @PiorGit, thank you very much for the suggestions 🙂!

I currently don't know if any of these would be possible, but you have to keep in mind that I can't change the unreal editor code, but rather "inject" things to it. If I were to change the editor codes, people would have to recompile it at every update (and it takes quite a few hours to be honest 😅)

Thought, it might be very interesting to investigate the possibilities of such changes 🙂.

I'm not sure about what you meant here, would you elaborate a bit?
> Editable input fields/highlighted text seem to behave in a non-standard way leading to many input errors when renaming things or even merely typing in values. I feel like this was designed in the interest of speed (reducing double clicks) but it leads to a very "unstable" behavior. As often with this sort of things, Blender tends to exhibit a much more robust behavior in that regard for instance.

In any case, it won't happen on Darker Nodes as it's only intended to change the theme, not the functionalities of the editor...

**Reply by PiorGit** [reply]
Hello !
Oh of course, I just thought I'd bring these up as a somewhat related topic. The "on click" vs "on release" is especially interesting I think.
Regarding the input fields : I will try to find the time to do a precise video comparaison when time allows. It's subtle stuff for sure !
